Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/DivXmlElementAccess.java

```diff
@@ -199,108 +158,19 @@ private final MdSec amdSecTypeOf(Mets mets, MdSecType mdSec) {
      *            type of meta-data section
      */
     private final void readMetadata(MdSecType mdSec, MdSec mdSecType) {
-        metadata.addAll(mdSec.getMdWrap().getXmlData().getAny().parallelStream().filter(JAXBElement.class::isInstance)
-                .map(JAXBElement.class::cast).map(JAXBElement::getValue).filter(KitodoType.class::isInstance)
-                .map(KitodoType.class::cast)
-                .flatMap(kitodoType -> Stream.concat(
-                    kitodoType.getMetadata().parallelStream()
-                            .map(metadataType -> new MetadataEntry(mdSecType, metadataType)),
-                    kitodoType.getMetadataGroup().parallelStream()
-                            .map(metadataGroupType -> new MetadataEntriesGroup(mdSecType, metadataGroupType))))
-                .collect(Collectors.toList()));
-    }
-
-    /**
-     * Returns the views associated with this structure.
-     * 
-     * @return the views
-     */
-    @Override
-    public List<AreaXmlElementAccessInterface> getAreas() {
-        return views;
-    }
-
-    /**
-     * Returns the substructures associated with this structure.
-     * 
-     * @return the substructures
-     */
-    @Override
-    public List<DivXmlElementAccessInterface> getChildren() {
-        return substructures;
-    }
-
-    /**
-     * Returns the label of this structure.
-     * 
-     * @return the label
-     */
-    @Override
-    public String getLabel() {
-        return label;
-    }
-
-    /**
-     * Returns the meta-data on this structure.
-     * 
-     * @return the meta-data
-     */
-    @Override
-    public Collection<MetadataAccessInterface> getMetadata() {
-        return metadata;
-    }
-
-    /**
-     * Returns the order label of this structure.
-     * 
-     * @return the order label
-     */
-    @Override
-    public String getOrderlabel() {
-        return orderlabel;
-    }
-
-    /**
-     * Returns the type of this structure.
-     * 
-     * @return the type
-     */
-    @Override
-    public String getType() {
-        return type;
-    }
-
-    /**
-     * Sets the label of this structure.
-     * 
-     * @param label
-     *            label to set
-     */
-    @Override
-    public void setLabel(String label) {
-        this.label = label;
-    }
-
-    /**
-     * Sets the order label of this structure.
-     * 
-     * @param orderlabel
-     *            order label to set
-     */
-    @Override
-    public void setOrderlabel(String orderlabel) {
-        this.orderlabel = orderlabel;
-    }
-
-    /**
-     * Sets the type of this structure.
-     * 
-     * @param type
-     *            type to set
-     */
-    @Override
-    public void setType(String type) {
-        this.type = type;
+        super.getMetadata()
```

Comment:
> There are a number of criteria that should make you consider not using a lambda:
> 
> Size The larger a lambda gets, the more difficult it makes it to follow the logic surrounding it.
> Repetition It's better to create a named function for repeated logic, although it's okay to repeat very simple lambdas that are spread apart.
> Naming If you can think of a great semantic name, you should use that instead, as it adds a lot of clarity to your code. I'm not talking names like priceIsOver100.  x -> x.price > 100 is just as clear as that name. I mean names like isEligibleVoter that replace a long list of conditions.
> Nesting Nested lambdas are really, really hard to read.
> 

https://softwareengineering.stackexchange.com/questions/341066/is-using-lambda-expressions-whenever-possible-in-java-good-practice

Reply:
This lambda is too long and totally unreadable...

Reply:
This is [already existing code](https://github.com/kitodo/kitodo-production/blob/master/Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/Structure.java#L202-L210). We can change it if you want to, but I would like to change it in a separate pull request because it is barely related to the current change (mainly formatting changes) which is already complicated enaugh.

I agree to that the lambdas are unreadable, which is due to our code formatter. I would like to write them totally different but I can’t, more like this:

    super.getMetadata()
    
    // get the value of the JAXB elements
    .map(JAXBElement.class::cast).map(JAXBElement::getValue)

    // all values of the JAXB elements are of kitodo type
    .filter(KitodoType.class::isInstance).map(KitodoType.class::cast)

    // convert to metadata entries and metadata groups
    .flatMap(kitodoType ->
        Stream.concat(
            kitodoType.getMetadata().parallelStream()
            .map(metadataType ->
                new MetadataXmlElementAccess(mdSecType, metadataType).getMetadataEntry()
            )
        ,
            kitodoType.getMetadataGroup().parallelStream()
            .map(metadataGroupType ->
                new MetadataGroupXmlElementAccess(mdSecType, metadataGroupType).getMetadataGroup()
            )
        )
    )
    .collect(Collectors.toList()));

If it is true that lambdas are *slower* in processing then this is in fact a point that we should change them. I used them because I believed they would be *faster* because they allow parallel processing.